zerver/models.py

```diff
@@ -214,8 +215,8 @@ def remote_user_to_email(remote_user):
 
 class RealmEmoji(models.Model):
     realm = models.ForeignKey(Realm)
-    name = models.TextField()
-    img_url = models.TextField()
+    name = models.TextField(validators=[MinLengthValidator(1), RegexValidator(regex=r'^[0-9a-zA-Z_\-\.]*$')])
```

Comment:
I will add a comment, even though I'm not sure this is the best regex. That's what we initially discussed, but I think it doesn't make much sense to allow punctuation at the end of the name and I think that would cause some issues with the requests in the frontend too (I think we don't encode the name of the emoji when deleting).